code/datums/loot_spawner.dm

```diff
@@ -144,77 +151,111 @@
 
 /datum/controller/subsystem/spawn_data/proc/get_spawn_value(npath)
 	var/atom/movable/A = npath
-	var/spawn_value = 10 * initial(A.spawn_frequency)/(initial(A.rarity_value) + log(10,max(get_spawn_price(A),1)))
-	return spawn_value
+	if(ispath(npath, /obj/item/weapon/gun))
+		return 10 * initial(A.spawn_frequency)/(initial(A.rarity_value)+(get_spawn_price(A)/GUN_PRICE_DIVISOR))
+	else if(ispath(npath, /obj/item/clothing))
+		return 10 * initial(A.spawn_frequency)/(initial(A.rarity_value) + (get_spawn_price(A)/CLOTH_PRICE_DIVISOR))
+	return 10 * initial(A.spawn_frequency)/(get_special_rarty_value(npath) + log(10,max(get_spawn_price(A),1)))//same from get_spawn_value()
+
+/datum/controller/subsystem/spawn_data/proc/get_special_rarty_value(npath)
```

Comment:
Could you do a doc-comment explaining what this function does, for future maintainers?

Reply:
> Could you do a doc-comment explaining what this function does, for future maintainers?

sure.